### ISSUE TITLE
all dimensions of model output is None

### ISSUE THREAD

**Issue by Ryuk17** [keras-team/keras#15963]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): pip 
- TensorFlow version (use command below): 2.0.1
- Python version: 3.7.4
- Bazel version (if compiling from source):
- GPU model and memory:
- Exact command to reproduce:

**Describe the problem**.

I have a model with two output and I use a generator, where feature is the input and gain, vad are labels
![image](https://user-images.githubusercontent.com/22525811/151288510-b2f7bc2e-7b10-46b1-be09-70bf9f0f0bc2.png)
![image](https://user-images.githubusercontent.com/22525811/151288980-e46fa95f-c300-491a-bdab-892ad5a255aa.png)

And I define a customer loss function
![image](https://user-images.githubusercontent.com/22525811/151288559-66b40ee7-648f-4949-b99a-ff044b1f14d8.png)
![image](https://user-images.githubusercontent.com/22525811/151288709-01aa560e-02a4-4b98-a580-90ae0784dcdc.png)

The the print results show that the dimensions of the model output is all None, which lead a Nan of loss
![image](https://user-images.githubusercontent.com/22525811/151288798-7e0d9f73-eaec-4463-a873-a9176e2b23c7.png)
![image](https://user-images.githubusercontent.com/22525811/151289128-73c87b73-3d1e-4f08-8975-c1d58ae98d3c.png)

I have no idea what cause this, any suggestions?

**Comment by Ryuk17** [comment]
I also check the output of generator, it's ok
![image](https://user-images.githubusercontent.com/22525811/151290388-576a77b3-54ca-47d4-a809-0828cdd5694d.png)
![image](https://user-images.githubusercontent.com/22525811/151290403-3bb0fbb4-4755-4800-b044-70ff8948b302.png)
